### PR TITLE
add text editor

### PR DESCRIPTION
**Pull request by aspasionov**
image
image
image

### CODE REVIEW

**Comment by mrnobody3** [comment]
looks good for me

File: src/screens/Home/index.jsx [hunk 1]
```diff
@@ -1,17 +1,130 @@
...
   const { user } = useAuth();
+  const { setAlert } = useAlert();
+
+  const client = new HttpClient();
+
+  const onChange = ({ target: { name, value, checked }}) => {
+    if(name === 'published') {
+      setPostData({ ...postData, [name]: checked })
+    } else {
+      setPostData({ ...postData, [name]: value })
```
Comment (SerhiienkoRuslan) [inline src/screens/Home/index.jsx:35]: setPostData({ ...postData, [name]: name === 'published' ? checked : value })

File: src/screens/Home/index.jsx [hunk 2]
```diff
@@ -1,17 +1,130 @@
...
+      if(error && message) {
+        await setAlert(message, 'error')
+      } else {
+        await setAlert('Post Created', 'success')
+      }
+      await fetchPosts();
+      await setModalVisible(false);
+
+    } catch (err) {
+      setLoading(false)
```
Comment (SerhiienkoRuslan) [inline src/screens/Home/index.jsx:68]: not necessary it here

File: src/screens/Home/index.jsx [hunk 3]
```diff
@@ -1,17 +1,130 @@
...
-    <div className='area h-[calc(100vh-37px)] p-2'>
-      <Decor/>
-      <h3>Home</h3>
-      {user && <h1>Hello {user?.username}</h1>}
-    </div>
-  )
+    <>
+      <div className='w-[90%] max-w-[1200px] mt-5 ml-auto mr-auto relative z-10'>
+        <div className='ml-auto w-44 border-b-2 border-b-gray-50 pb-2 px-2 mb-2'>
+          <Btn onClick={() => setModalVisible(true)}>
```
Comment (SerhiienkoRuslan) [inline src/screens/Home/index.jsx:82]: Move btn and modal component to separate component

File: src/screens/Home/components/Post/index.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import Modal from 'components/Modal';
+import { NavLink } from 'react-router-dom';
+import PropTypes from 'prop-types'
+import parse from 'html-react-parser';
+import { UserIcon } from '@heroicons/react/solid'
+
+const Post = ({ post }) => {
+  const [visibleModal, setModalVisible] = React.useState(false)
```
Comment (nester7x) [inline src/screens/Home/components/Post/index.jsx:9]: setVisibleModal

File: src/screens/Home/index.jsx [hunk 5]
```diff
@@ -1,17 +1,130 @@
...
+const defaultPost = {
+  title: '',
+  content: '',
+  published: false
+}
 
 const Home = () => {
+  const [posts, setPosts] = React.useState([]);
+  const [loading, setLoading] = React.useState(false);
+  const [modalVisible, setModalVisible] = React.useState(false);
```
Comment (nester7x) [inline src/screens/Home/index.jsx:23]: setVisibleModal

File: src/screens/Home/index.jsx [hunk 6]
```diff
@@ -1,17 +1,130 @@
 import React from 'react';
+import HttpClient from 'api/base.api';
 import useAuth from "hooks/useAuth";
-import Decor from 'components/Decorations';
+import useAlert from "hooks/useAlert";
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
```
Comment (nester7x) [inline src/screens/Home/index.jsx:6]: make CKEditor as a component

File: src/routes/index.jsx [hunk 7]
```diff
@@ -5,7 +5,6 @@ import ProtectedRoutes from './ProtectedRoutes';
 import Home from '../screens/Home';
 import Login from '../screens/Login';
 import Chat from '../screens/Chat';
-import Posts from '../screens/Posts';
 import EditProfile from '../screens/EditProfile';
 import User from '../screens/User';
```
Comment (nester7x) [inline src/routes/index.jsx:9]: try to use absolute path ('screens/Home')

File: src/index.css [hunk 8]
```diff
@@ -188,7 +188,7 @@ a {
     border-radius: 13px;
     overflow: hidden;
     z-index: 9999999999999999999999998;
```
Comment (SerhiienkoRuslan) [inline src/index.css:190]: wtf

**Review by SerhiienkoRuslan — APPROVED** [review]

**Review by nester7x — APPROVED** [review]

**Pull request merged**